compiler/rustc_ast_passes/src/errors.rs

```diff
@@ -509,17 +509,25 @@ pub struct WhereClauseBeforeTypeAlias {
 }
 
 #[derive(Subdiagnostic)]
-#[multipart_suggestion(
-    ast_passes_suggestion,
-    applicability = "machine-applicable",
-    style = "verbose"
-)]
-pub struct WhereClauseBeforeTypeAliasSugg {
-    #[suggestion_part(code = "")]
-    pub left: Span,
-    pub snippet: String,
-    #[suggestion_part(code = "{snippet}")]
-    pub right: Span,
+
+pub enum WhereClauseBeforeTypeAliasSugg {
```

Comment:
The split into `Remove` & `Move` was not only done to provide a more accurate diagnostic message for the edge case “two where clauses: empty leading one and a trailing one” but also to prevent the debug assertion *empty span and no suggestion* from firing.